docs/concepts/resource_flavor.md

```diff
@@ -0,0 +1,92 @@
+# ResourceFlavor
+
+Resources in a cluster are typically not homogeneous. Resources could differ in:
+- pricing and availability (for example, spot verus on-demand VMs)
+- architecture (for example, x86 versus ARM CPUs)
+- brands and models (for example, Radeon 7000 versus Nvidia A100 versus T4 GPUs)
+
+A ResourceFlavor is an object that represents these resource variations and
+allows you to associate them with cluster nodes through labels and taints.
+
+**Note**: If your the resources in your cluster are homogeneous, you can use 
+an [empty ResourceFlavor](#empty-resourceflavor) instead of adding labels to custom ResourceFlavors.
+
+A sample ResourceFlavor looks like the following:
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: ResourceFlavor
+metadata:
+  name: spot
+nodeSelector:
+  instance-type: spot
+taints:
+- effect: NoSchedule
+  key: spot
+  value: "true"
+```
+
+You can use the `.metadata.name` field to reference a ResourceFlavor from a
+[ClusterQueue](/docs/concepts/cluster_queue.md) in the `.spec.resources[*].flavors[*].name` field.
+
+## ResourceFlavor labels
+
+**Requires Kubernetes 1.23 or newer**
+
+To associate a ResourceFlavor with a subset of nodes of your cluster, you can
+configure the `.nodeSelector` field with matching node labels that uniquely identify
+the nodes. If you are using [cluster autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler)
+(or equivalent controllers), make sure that the controller is configured to add those labels when
+adding new nodes.
+
+To guarantee that the workload Pods run on the nodes associated to the flavor
```

Comment:
Maybe we can uppercase workload to `Workload` across the document to mark it a special Object, just like ResourceFlavor and ClusterQueue.